File: 0118 pascals-triangle/solution.java
class Solution {
    public List<List<Integer>> generate(int n) {
        List<List<Integer>> ans = new ArrayList<>();
        for(int i=0;i<n;i++) {
            List<Integer> ians = new ArrayList<>();
            for(int j=0;j<=i;j++) {
            if(j==0 || j==i) ians.add(1);
            else {
                ians.add(ans.get(i-1).get(j)+ans.get(i-1).get(j-1));
            }
            }
            ans.add(ians);
        }
        return ans;
    }
}

File: 0118 pascals-triangle/solution.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans;
        for(int i=0;i<numRows;i++) {
            vector<int> li;
            for(int j=0;j<=i;j++) {
                if(j==0 || j==i) li.push_back(1);
                else {
                    int num = ans[i-1][j] + ans[i-1][j-1];
                    li.push_back(num);
                }
            }
            ans.push_back(li);
        }
        return ans;
    }
};
